typings/index.d.ts

```diff
@@ -146,6 +146,9 @@ export interface Actions {
   ) => void
   clearSelection: (cb?: Function) => void
   clearItems: () => void
+  setItemCount: (count: number) => void
+  unsetItemCount: () => void
+  setState: (stateToSet: object, cb?: Function) => void
```

Comment:
Could probably make this more strict. I don't know TypeScript, but the possible values are:

```javascript
{
  type: string,
  isOpen: boolean,
  inputValue: string,
  selectedItem: any,
  highlightedIndex: number | null,
}
```

Reply:
Agreed. I've also missed the fact that `stateToSet` can be a function.
Looking at the implementations for the actions with a `otherStateToSet` argument, and I think all the `otherStateToSet?: object` types could be stricter, too, perhaps using `Partial<StateChangeOptions>`.

Reply:
Talk to @stereobooster. They've done some work in the `next` branch. We're preparing for downshift v2.0.0. :)

Reply:
This is the PR Kent talking about https://github.com/paypal/downshift/pull/373/files.

If you want to take into account those changes you need to create branch based on `next`.